<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>HITChip tools</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">HITChip tools</h1>



<!--
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{microbiome tutorial - hitchip}
  %\usepackage[utf8]{inputenc}
  %\VignetteEncoding{UTF-8}  
-->
<div id="hitchip-and-other-phylogenetic-microarrays" class="section level2">
<h2>HITChip and other phylogenetic microarrays</h2>
<ul>
<li><a href="https://github.com/microbiome/HITChipDB/blob/master/index./index.html">Extracting data from HITChip database</a></li>
<li><a href="Probelevel.html">Probe level studies (phylogenetic microarrays)</a></li>
</ul>
<div id="importing-hitchip-data-to-phyloseq-format" class="section level3">
<h3>Importing HITChip data to phyloseq format</h3>
<p>Define the data folder.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Define example data path (replace here data.directory with your own path)</span>
<span class="kw">library</span>(microbiome)
data.directory &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;microbiome&quot;</span>)
<span class="kw">print</span>(data.directory)</code></pre></div>
<pre><code>## [1] &quot;/tmp/Rtmpg7PRbU/Rinst3cd646ca5b02/microbiome/extdata&quot;</code></pre>
</div>
</div>
<div id="installing-hitchipdb-package" class="section level2">
<h2>Installing HITChipDB package</h2>
<p>The HITChipDB package contains additional routines to fetch and preprocess HITChip (or MIT/PITChip) data from the MySQL database. Note that this package is <strong>not needed by most users</strong> and the data is protected by password/IP combinations. Ask details from admins. Install the package in R with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(microbiome)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Install additional dependencies</span>
<span class="kw">source</span>(<span class="st">&quot;http://www.bioconductor.org/biocLite.R&quot;</span>)

<span class="kw">biocLite</span>(<span class="st">&quot;DBI&quot;</span>)
<span class="kw">biocLite</span>(<span class="st">&quot;RPA&quot;</span>)
<span class="kw">biocLite</span>(<span class="st">&quot;svDialogs&quot;</span>)

<span class="kw">library</span>(devtools) <span class="co"># Load the devtools package</span>
<span class="kw">install_github</span>(<span class="st">&quot;microbiome/HITChipDB&quot;</span>) <span class="co"># Install the package</span>
<span class="co"># Also install RMySQL, multicore and tcltk !</span>
<span class="kw">source</span>(<span class="st">&quot;http://www.bioconductor.org/biocLite.R&quot;</span>)
<span class="kw">biocLite</span>(<span class="st">&quot;RMySQL&quot;</span>) <span class="co"># multicore, tcltk?</span>
<span class="co"># Test installation by loading the microbiome package in R</span>
<span class="kw">library</span>(<span class="st">&quot;HITChipDB&quot;</span>)</code></pre></div>
<p>With HITChip, <a href="http://www.computer.org/csdl/trans/tb/2011/01/ttb2011010217-abs.html">fRPA</a> is the recommended preprocessing method. You can add new metadata fields in the template metadata file in your HITChip data folder and exporting it again to tab-separated .tab format. Some standard, self-explanatory field names include ‘sample’, ‘time’, ‘subject’, ‘group’, ‘gender’, ‘diet’, ‘age’. You can leave these out or include further fields. Import HITChip phylotype-level data in <a href="https://github.com/joey711/phyloseq">phyloseq</a> format (note: the precalculated matrices are calculated with detection = 0):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(HITChipDB)
pseq &lt;-<span class="st"> </span>HITChipDB::<span class="kw">read_hitchip</span>(data.directory, <span class="dt">method =</span> <span class="st">&quot;frpa&quot;</span>)$pseq</code></pre></div>
<p>Get higher taxonomic levels, use (on HITChip we use L1/L2 instead of Phylum/Genus):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pseq.L2 &lt;-<span class="st"> </span><span class="kw">aggregate_taxa</span>(pseq, <span class="dt">level =</span> <span class="st">&quot;L2&quot;</span>)
pseq.L1 &lt;-<span class="st"> </span><span class="kw">aggregate_taxa</span>(pseq, <span class="dt">level =</span> <span class="st">&quot;L1&quot;</span>)</code></pre></div>
<p>Importing HITChip probe-level data and taxonomy from HITChip output directory (these are not available in the phyloseq object):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">probedata &lt;-<span class="st"> </span>HITChipDB::<span class="kw">read_hitchip</span>(data.directory, <span class="dt">method =</span> <span class="st">&quot;frpa&quot;</span>)$probedata
taxonomy.full &lt;-<span class="st"> </span>HITChipDB::<span class="kw">read_hitchip</span>(data.directory, <span class="dt">method =</span> <span class="st">&quot;frpa&quot;</span>)$taxonomy.full</code></pre></div>
<p>Convert your own data into phyloseq format as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># We need to choose the HITChip data level to be used in the analyses</span>
<span class="co"># In this example use HITChip L2 data (note: this is in absolute scale)</span>
res &lt;-<span class="st"> </span><span class="kw">read_hitchip</span>(<span class="dt">method =</span> <span class="st">&quot;frpa&quot;</span>, <span class="dt">data.dir =</span> data.directory)

<span class="co"># Species-level data matrix</span>
otu &lt;-<span class="st"> </span><span class="kw">abundances</span>(res$pseq)@.Data 

<span class="co"># Corresponding sample metadata</span>
meta &lt;-<span class="st"> </span>res$meta

<span class="co"># Taxonomy</span>
<span class="co"># First get an experimental function from the microbiome package</span>
f &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;inst/extdata/get_hitchip_taxonomy.R&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;microbiome&quot;</span>)
<span class="kw">source</span>(f)
taxonomy &lt;-<span class="st"> </span><span class="kw">get_hitchip_taxonomy</span>(<span class="st">&quot;HITChip&quot;</span>, <span class="st">&quot;filtered&quot;</span>)
taxonomy &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">as.data.frame</span>(taxonomy[, <span class="kw">c</span>(<span class="st">&quot;L1&quot;</span>, <span class="st">&quot;L2&quot;</span>, <span class="st">&quot;species&quot;</span>)]))
<span class="kw">rownames</span>(taxonomy) &lt;-<span class="st"> </span><span class="kw">as.vector</span>(taxonomy[, <span class="st">&quot;species&quot;</span>])

<span class="co"># Merging data matrices into phyloseq format:</span>
pseq &lt;-<span class="st"> </span>HITChipDB::<span class="kw">hitchip2physeq</span>(<span class="kw">t</span>(otu), meta, taxonomy)</code></pre></div>
<div id="refs" class="references">

</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
